File: gencolors.py
import sys
import math

# https://twitter.com/jon_barron/status/1388233935641976833
# https://basecase.org/env/on-rainbows
# https://krazydad.com/tutorials/makecolors.php
num_colors = int(sys.argv[1])
f = lambda x: int(math.sin(math.pi * x)**2 * 230)
for i in range(0, num_colors):
    h = i / num_colors
    r = f(3/6-h); g = f(5/6-h); b = f(7/6-h)
    print("{}: {:02X}{:02X}{:02X}".format(i, r, g, b))

File: tools/gencolors.js
"use strict";

// https://twitter.com/jon_barron/status/1388233935641976833
// https://basecase.org/env/on-rainbows
// https://krazydad.com/tutorials/makecolors.php
let numColors = process.argv[2];
let f = x => Math.floor(Math.sin(Math.PI * x)**2 * 230).toString(16).toUpperCase().padStart(2, '0');
for (let i = 0; i < numColors; i++) {
    let h = i / numColors;
    let r = f(3/6 - h), g = f(5/6 - h), b = f(7/6 - h);
    console.log(`${i}: ${r}${g}${b}`);
}
